File: test-dagger.go
package main

import (
	"context"
	"fmt"
	"log"

	"dagger.io/dagger"
)

func main() {
	ctx := context.Background()

	// Initialize Dagger client
	client, err := dagger.Connect(ctx, dagger.WithLogOutput(nil))
	if err != nil {
		log.Fatalf("Failed to connect to Dagger: %v", err)
	}
	defer client.Close()

	fmt.Println("Testing basic container creation...")
	
	// Test basic container creation
	container := client.Container().From("python:3.11-slim")

	fmt.Println("Testing command execution...")
	
	// Test command execution
	result, err := container.WithExec([]string{"python", "--version"}).Stdout(ctx)
	if err != nil {
		log.Fatalf("Failed to execute command: %v", err)
	}
	fmt.Printf("Python version: %s", result)

	fmt.Println("Testing file operations...")
	
	// Test file operations
	containerWithFile := container.WithNewFile("/hello.py", "print('Hello from Dagger!')")
	output, err := containerWithFile.WithExec([]string{"python", "/hello.py"}).Stdout(ctx)
	if err != nil {
		log.Fatalf("Failed to execute script: %v", err)
	}
	fmt.Printf("Script output: %s", output)

	fmt.Println("✅ All Dagger tests passed!")
}

File: test-dagger.py
#!/usr/bin/env python3

import sys
import dagger
import asyncio

async def main():
    async with dagger.Connection() as client:
        # Test basic container creation
        print("Testing basic container creation...")
        container = client.container().from_("python:3.11-slim")
        
        # Test command execution
        print("Testing command execution...")  
        result = await container.with_exec(["python", "--version"]).stdout()
        print(f"Python version: {result.strip()}")
        
        # Test file operations
        print("Testing file operations...")
        container_with_file = container.with_new_file("/hello.py", "print('Hello from Dagger!')")
        output = await container_with_file.with_exec(["python", "/hello.py"]).stdout()
        print(f"Script output: {output.strip()}")
        
        print("✅ All Dagger tests passed!")

if __name__ == "__main__":
    asyncio.run(main())